### PR TITLE
feat(ENG-1609): fetch web platform instead of having it in repo

### PR DESCRIPTION
**Pull request by itayd**
- fetch according to `web/webplatform/VERSION`.
- display version in up.
- release and docker builds are handled as well.

### CODE REVIEW

File: internal/backend/svc/banner.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ var banner string
 
 var bannerTemplate = template.Must(template.New("banner").Parse(banner))
 
-func printBanner(cfg *bannerConfig, opts RunOptions, addr, wpAddr, temporalFrontendAddr, temporalUIAddr string) {
+func printBanner(cfg *bannerConfig, opts RunOptions, addr, wpAddr, wpVersion, temporalFrontendAddr, temporalUIAddr string) {
```
Comment (daabr) [inline internal/backend/svc/banner.go:30]: You've officially crossed the threshold of "too many parameters, better use a struct".

Also, "addr" and "wp*" are unclear param names.

Also, please document this function, and mention the boolean config key to control showing this banner (`banner.show`).

I'm ok with doing this in a separate PR, but if so please do it soon-ish.

File: internal/backend/svc/banner.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 
-        _..--~~~~--.._             autokitteh   {{ .Version }}
-     _--              --_          PID:         {{ .PID }}
-    '                    `         gRPC/HTTP:   {{ .Addr }}
-   '                      `        Web:         {{ .WebAddr }}
+        _..--~~~~--.._             autokitteh    {{ .Version }}
+     _--              --_          PID:          {{ .PID }}
+    '                    `         gRPC/HTTP:    {{ .Addr }}
+   '                      `        Web{{ .WebVer }}  {{ .WebAddr }}
```
Comment (daabr) [inline internal/backend/svc/banner.txt:5]: It's very easy to make this look messed up when you inject a variable-length string in the left side, especially with no space after "Web", and no ":" before the address. Maybe `Web: address (version)` instead? Or separate "Web Address" and "Web Version"?

File: web/webplatform/VERSION [hunk 3]
```diff
@@ -0,0 +1 @@
+2.29.1
```
Comment (daabr) [inline web/webplatform/VERSION:1]: Question: any plan to make this dynamic?

File: web/webplatform/nothing.zip [hunk 4]
```diff
@@ -0,0 +1 @@
+This is not really a zip file, just a placeholder to prevent go embed from failing if no distribution was downloaded.
```
Comment (daabr) [inline web/webplatform/nothing.zip:1]: But shouldn't it fail in this case?

File: web/webplatform/fetch.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    exit 0
+fi
+
+echo "not downloaded yet."
+
+PREV="$(ls -1 autokitteh-web-v*.zip > /dev/null 2>&1 || true)"
+
+echo "fetching ${FILENAME}..."
+
+curl -sL "https://github.com/autokitteh/web-platform/releases/download/v${VERSION}/${FILENAME}" > "${FILENAME}"
```
Comment (daabr) [inline web/webplatform/fetch.sh:23]: Can you also read and check the file's hash?

File: web/webplatform/VERSION (hunk 3 above)
Comment (itayd) [reply web/webplatform/VERSION:1]: nope. we need to be explicit.

File: web/webplatform/nothing.zip (hunk 4 above)
Comment (itayd) [reply web/webplatform/nothing.zip:1]: nope, sometimes web platform is not required.

File: web/webplatform/VERSION (hunk 3 above)
Comment (daabr) [reply web/webplatform/VERSION:1]: ok, then please add a document in the catcafe to clarify who does what, how, and when / how often.

File: web/webplatform/fetch.sh (hunk 5 above)
Comment (itayd) [reply web/webplatform/fetch.sh:23]: doesn't look like we publish the hash on the release, we'll have to do that. @RonenMars ?

Comment (RonenMars) [reply web/webplatform/fetch.sh:23]: It's possible, and I'll implement it on my side.

Do we want the hash to be part of the file name or a separate text file?
I guess a separate text file would be a better option, right?

Comment (itayd) [reply web/webplatform/fetch.sh:23]: Separate text file. see https://github.com/temporalio/temporal/releases/tag/v1.25.1 for example, file `checksums.txt` which is the standard i think

**Review by daabr — APPROVED** [review]

File: internal/backend/svc/banner.go (hunk 1 above)
Comment (itayd) [reply internal/backend/svc/banner.go:30]: in a futute pr.

File: internal/backend/svc/banner.txt (hunk 2 above)
Comment (itayd) [reply internal/backend/svc/banner.txt:5]: in a futute pr.

**Pull request merged**